src/fitutil/SystematicManager.cpp

```diff
@@ -1,39 +1,196 @@
         return;
 
-    // construct the response matricies
-    for(size_t i = 0; i < fSystematics.size(); i++)
-        fSystematics[i] -> Construct();
+    //Construct the response matrices.
+    for (std::map<std::string,std::vector<Systematic*> >::const_iterator group=fGroups.begin(); group!= fGroups.end(); ++group ) {
+        for(size_t i = 0; i < group->second.size(); i++)
+            fGroups[group->first].at(i) -> Construct();
+    }
 
-    
-    // Assemble the overall response
-    fTotalResponse = fSystematics.at(0) -> GetResponse();
-    for(size_t i = 1; i < fSystematics.size(); i++)
-      fTotalResponse *= fSystematics.at(i) -> GetResponse();
+    //This loop should construct all fGroups.
+    for (std::map<std::string,std::vector<Systematic*> >::const_iterator group =fGroups.begin(); group != fGroups.end(); ++group ) {
+        SparseMatrix resp = fGroups[group->first].at(0) -> GetResponse();
+        for(size_t i = 1; i < group->second.size(); i++)
+            resp *= fGroups[group->first].at(i) -> GetResponse();
+        fTotalReponses[group->first]=resp;
+    }
 }
 
 const SparseMatrix&
-SystematicManager::GetTotalResponse() const{
-     return fTotalResponse;
+SystematicManager::GetTotalResponse(const std::string& groupName_) const{
+    //should you construct?
+    try{
+        return fTotalReponses.at(groupName_);
+    }
+    catch(const std::out_of_range& e_){
+        throw NotFoundError(Formatter()<<
+                "SystematicManager :: name : "<< groupName_<<
+                " not found in systematic map");
+    }
+}
+
+
+void
+SystematicManager::Add(Systematic* sys_, const std::string& groupName_){
+    fGroups[groupName_].push_back(sys_);
+    fNGroups = fGroups.size();
+}
+
+void
+SystematicManager::UniqueSystematics(const std::vector<std::string>& syss_){
+    std::vector<std::string> allname;
+    for (int i = 0; i <syss_.size(); ++i) {
+        if(fGroups.find( syss_.at(i) ) == fGroups.end())
+            throw NotFoundError (
+                    Formatter()<<"SystematicManager:: Systematic group "<<
+                    syss_.at(i)<<
+                    " not found in complete set of groups. " );
+
+        std::vector<Systematic*> group = fGroups[syss_.at(i)];
+
+        for (int j = 0; j < group.size(); ++j) {
+            if(std::find(allname.begin(),allname.end(), group.at(j)->GetName() ) == allname.end())
+                allname.push_back(group.at(j)->GetName());
+            else
+                throw NotFoundError(Formatter()<<
+                        "SystematicManager :: Systematic: "<< 
+                        group.at(i)->GetName()<<
+                        " is in more than one group. BAD!!!");
+        } 
+    }
 }
-                                        
-void 
-SystematicManager::Add(Systematic* systematic_){    
-    fSystematics.push_back(systematic_);
-    fNSystematics++;
+
+void
+SystematicManager::AddDist(const BinnedED& pdf, const std::vector<std::string>& syss_){
```

Comment:
If you add a distribution like this it can easily not be in default right?
`AddDist(mydist, {"somethingOtherThanDefault"})`

I know the code correctly applies it anyway, but I'm slightly concerned by the output of this code (not tested):

`sysMan.AddDist(mydist, {"somethingOtherThanDefault"}) // *doesn't* get the string "default" in its vector of groups`
`sysMan.AddDist(mydist2) // *does* get the string "default" in its vector of groups`

If someone adds code to get the groups for a given distribution then the output will suggest that somethings don't get the default systematics even though they did. 

That might sound like the ramblings of a mad man -- let me know